apps/tailwind-components/components/input/Tree.vue

```diff
@@ -0,0 +1,167 @@
+  }
+}
+
+function expandSelection(node: ITreeNode) {
+  let selection: string[] = [];
+  node.children.forEach((child) => {
+    selection.push(child.name);
+    if (child.children) {
+      selection = [...selection, ...expandSelection(child)];
+    }
+  });
+
+  return selection;
+}
+
+function toggleSelect(node: ITreeNode) {
+  if (props.modelValue.includes(node.name)) {
+    // remove node from selected nodes and emit new model
+    emit(
+      "update:modelValue",
+      props.modelValue.filter((n) => n !== node.name)
+    );
+  } else {
+    let currnetSelection = [...props.modelValue, node.name];
+    if (props.expandSelected) {
+      currnetSelection = [...currnetSelection, ...expandSelection(node)];
+    }
+    const deduplicated = [...new Set(currnetSelection)];
+    // clone model and add new before emitting
+    emit("update:modelValue", deduplicated);
+  }
+}
+
+function handleChildSelect(selected: string[], parent: ITreeNode) {
+  const siblingNames = parent.children.map((n) => n.name);
+  const selectParent =
+    siblingNames.some((siblingName) => selected.includes(siblingName)) &&
+    siblingNames.every((siblingName) => selected.includes(siblingName));
+  const updatedSelection = selectParent ? [...selected, parent.name] : selected;
+
+  const deduplicated = [...new Set(updatedSelection)];
+  emit("update:modelValue", deduplicated);
+}
+</script>
+
+<template>
+  <ul
+    :class="[
+      inverted
+        ? 'text-search-filter-group-title-inverted'
+        : 'text-search-filter-group-title',
+    ]"
+  >
+    <li v-for="node in nodes" :key="node.name" class="mt-2.5 relative">
+      <span class="flex items-center">
+        <span
+          v-if="node.children?.length"
+          @click.stop="toggleExpand(node.name)"
+          class="-left-[11px] top-0 rounded-full hover:cursor-pointer h-6 w-6 flex items-center justify-center absolute z-20"
+          :class="[
+            inverted
+              ? 'text-search-filter-group-toggle-inverted hover:bg-search-filter-group-toggle-inverted'
+              : 'text-search-filter-group-toggle hover:bg-search-filter-group-toggle',
+          ]"
+        >
+          <BaseIcon
+            :name="
+              expandedNodes.includes(node.name) ? 'caret-down' : 'caret-up'
+            "
+            :width="20"
+          />
+        </span>
+        <template v-if="!isRoot">
+          <BaseIcon
+            v-if="node.children?.length"
+            name="collapsible-list-item-sub"
+            :width="20"
+            class="text-blue-200 absolute -top-[9px]"
+          />
+          <BaseIcon
+            v-else
+            name="collapsible-list-item"
+            :width="20"
+            class="text-blue-200 absolute -top-[9px]"
+          />
+        </template>
+      </span>
+      <div class="flex items-start ml-4">
+        <div class="flex items-center">
+          <input
+            type="checkbox"
+            :indeterminate="
```

Comment:
The indeterminate status seems to stop at the third level.

Reply:
i'll try and reproduce the issue , if im not able to i'll ask for some steps to reproduce this behaviour